.github/workflows/ci.yml

```diff
@@ -38,8 +38,70 @@ jobs:
           ${{ runner.os }}-${{ matrix.python-version }}-pip-
     - name: Install dependencies
       run: |
-        python -m pip install .
-        python -m pip install -r test-requirements.txt
+        python -m pip install --editable .[test]
     - name: Run tests
-      run: |
-        python -m pytest tests
+      run: tox
+
+  publish-docker-image:
+    runs-on: ubuntu-latest
+    needs: [unit-tests]
+    steps:
+      - name: System Dependencies for Packaging
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y python3-pip
+          sudo pip install --upgrade pip
+          sudo pip install yq
```

Comment:
Are these necessary? Also, why do we need sudo here?

Reply:
`yq` contains `tomlq` and is a wrapper around `jq` which makes it very easy to extract meta-data from a TOML file. I use this to fill-in docker labels during build. This allows us to keep all that meta-data in one single place (the `pyproject.toml`).

Instead of `sudo` I can switch to a venv instead. I will still need sudo to install the `python3-venv` package then. Since Python 3.11, `pip install` raises an error, even when doing a "user-install". So a venv will be necessary. I will make that change and leave this thread unresolved in case you have additional feedback.

Reply:
Changes to using a venv has been applied.